// @aws-sdk/client-redshift/dist-es/commands/AuthorizeDataShareCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { AuthorizeDataShareMessageFilterSensitiveLog, DataShareFilterSensitiveLog, } from "../models/models_0";
import { deserializeAws_queryAuthorizeDataShareCommand, serializeAws_queryAuthorizeDataShareCommand, } from "../protocols/Aws_query";
var AuthorizeDataShareCommand = (function (_super) {
    __extends(AuthorizeDataShareCommand, _super);
    function AuthorizeDataShareCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    AuthorizeDataShareCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "RedshiftClient";
        var commandName = "AuthorizeDataShareCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: AuthorizeDataShareMessageFilterSensitiveLog,
            outputFilterSensitiveLog: DataShareFilterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    AuthorizeDataShareCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryAuthorizeDataShareCommand(input, context);
    };
    AuthorizeDataShareCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryAuthorizeDataShareCommand(output, context);
    };
    return AuthorizeDataShareCommand;
}($Command));
export { AuthorizeDataShareCommand };
