// @aws-sdk/client-redshift/dist-es/commands/CancelResizeCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { CancelResizeMessageFilterSensitiveLog, ResizeProgressMessageFilterSensitiveLog, } from "../models/models_0";
import { deserializeAws_queryCancelResizeCommand, serializeAws_queryCancelResizeCommand } from "../protocols/Aws_query";
var CancelResizeCommand = (function (_super) {
    __extends(CancelResizeCommand, _super);
    function CancelResizeCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    CancelResizeCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "RedshiftClient";
        var commandName = "CancelResizeCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: CancelResizeMessageFilterSensitiveLog,
            outputFilterSensitiveLog: ResizeProgressMessageFilterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    CancelResizeCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryCancelResizeCommand(input, context);
    };
    CancelResizeCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryCancelResizeCommand(output, context);
    };
    return CancelResizeCommand;
}($Command));
export { CancelResizeCommand };
