// @aws-sdk/client-redshift/dist-es/commands/CreateEndpointAccessCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { CreateEndpointAccessMessageFilterSensitiveLog, EndpointAccessFilterSensitiveLog, } from "../models/models_0";
import { deserializeAws_queryCreateEndpointAccessCommand, serializeAws_queryCreateEndpointAccessCommand, } from "../protocols/Aws_query";
var CreateEndpointAccessCommand = (function (_super) {
    __extends(CreateEndpointAccessCommand, _super);
    function CreateEndpointAccessCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    CreateEndpointAccessCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "RedshiftClient";
        var commandName = "CreateEndpointAccessCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: CreateEndpointAccessMessageFilterSensitiveLog,
            outputFilterSensitiveLog: EndpointAccessFilterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    CreateEndpointAccessCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryCreateEndpointAccessCommand(input, context);
    };
    CreateEndpointAccessCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryCreateEndpointAccessCommand(output, context);
    };
    return CreateEndpointAccessCommand;
}($Command));
export { CreateEndpointAccessCommand };
