// @aws-sdk/client-redshift/dist-es/commands/CreateHsmConfigurationCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { CreateHsmConfigurationMessageFilterSensitiveLog, CreateHsmConfigurationResultFilterSensitiveLog, } from "../models/models_0";
import { deserializeAws_queryCreateHsmConfigurationCommand, serializeAws_queryCreateHsmConfigurationCommand, } from "../protocols/Aws_query";
var CreateHsmConfigurationCommand = (function (_super) {
    __extends(CreateHsmConfigurationCommand, _super);
    function CreateHsmConfigurationCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    CreateHsmConfigurationCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "RedshiftClient";
        var commandName = "CreateHsmConfigurationCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: CreateHsmConfigurationMessageFilterSensitiveLog,
            outputFilterSensitiveLog: CreateHsmConfigurationResultFilterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    CreateHsmConfigurationCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryCreateHsmConfigurationCommand(input, context);
    };
    CreateHsmConfigurationCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryCreateHsmConfigurationCommand(output, context);
    };
    return CreateHsmConfigurationCommand;
}($Command));
export { CreateHsmConfigurationCommand };
