// @aws-sdk/client-redshift/dist-es/commands/CreateSnapshotScheduleCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { CreateSnapshotScheduleMessageFilterSensitiveLog, SnapshotScheduleFilterSensitiveLog, } from "../models/models_0";
import { deserializeAws_queryCreateSnapshotScheduleCommand, serializeAws_queryCreateSnapshotScheduleCommand, } from "../protocols/Aws_query";
var CreateSnapshotScheduleCommand = (function (_super) {
    __extends(CreateSnapshotScheduleCommand, _super);
    function CreateSnapshotScheduleCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    CreateSnapshotScheduleCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "RedshiftClient";
        var commandName = "CreateSnapshotScheduleCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: CreateSnapshotScheduleMessageFilterSensitiveLog,
            outputFilterSensitiveLog: SnapshotScheduleFilterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    CreateSnapshotScheduleCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryCreateSnapshotScheduleCommand(input, context);
    };
    CreateSnapshotScheduleCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryCreateSnapshotScheduleCommand(output, context);
    };
    return CreateSnapshotScheduleCommand;
}($Command));
export { CreateSnapshotScheduleCommand };
