// @aws-sdk/client-redshift/dist-es/commands/CreateAuthenticationProfileCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { CreateAuthenticationProfileMessageFilterSensitiveLog, CreateAuthenticationProfileResultFilterSensitiveLog, } from "../models/models_0";
import { deserializeAws_queryCreateAuthenticationProfileCommand, serializeAws_queryCreateAuthenticationProfileCommand, } from "../protocols/Aws_query";
var CreateAuthenticationProfileCommand = (function (_super) {
    __extends(CreateAuthenticationProfileCommand, _super);
    function CreateAuthenticationProfileCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    CreateAuthenticationProfileCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "RedshiftClient";
        var commandName = "CreateAuthenticationProfileCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: CreateAuthenticationProfileMessageFilterSensitiveLog,
            outputFilterSensitiveLog: CreateAuthenticationProfileResultFilterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    CreateAuthenticationProfileCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryCreateAuthenticationProfileCommand(input, context);
    };
    CreateAuthenticationProfileCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryCreateAuthenticationProfileCommand(output, context);
    };
    return CreateAuthenticationProfileCommand;
}($Command));
export { CreateAuthenticationProfileCommand };
