// @aws-sdk/client-redshift/dist-es/commands/CreateClusterParameterGroupCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { CreateClusterParameterGroupMessageFilterSensitiveLog, CreateClusterParameterGroupResultFilterSensitiveLog, } from "../models/models_0";
import { deserializeAws_queryCreateClusterParameterGroupCommand, serializeAws_queryCreateClusterParameterGroupCommand, } from "../protocols/Aws_query";
var CreateClusterParameterGroupCommand = (function (_super) {
    __extends(CreateClusterParameterGroupCommand, _super);
    function CreateClusterParameterGroupCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    CreateClusterParameterGroupCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "RedshiftClient";
        var commandName = "CreateClusterParameterGroupCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: CreateClusterParameterGroupMessageFilterSensitiveLog,
            outputFilterSensitiveLog: CreateClusterParameterGroupResultFilterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    CreateClusterParameterGroupCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryCreateClusterParameterGroupCommand(input, context);
    };
    CreateClusterParameterGroupCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryCreateClusterParameterGroupCommand(output, context);
    };
    return CreateClusterParameterGroupCommand;
}($Command));
export { CreateClusterParameterGroupCommand };
