// @aws-sdk/client-redshift/dist-es/commands/CreateUsageLimitCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { CreateUsageLimitMessageFilterSensitiveLog, UsageLimitFilterSensitiveLog, } from "../models/models_0";
import { deserializeAws_queryCreateUsageLimitCommand, serializeAws_queryCreateUsageLimitCommand, } from "../protocols/Aws_query";
var CreateUsageLimitCommand = (function (_super) {
    __extends(CreateUsageLimitCommand, _super);
    function CreateUsageLimitCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    CreateUsageLimitCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "RedshiftClient";
        var commandName = "CreateUsageLimitCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: CreateUsageLimitMessageFilterSensitiveLog,
            outputFilterSensitiveLog: UsageLimitFilterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    CreateUsageLimitCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryCreateUsageLimitCommand(input, context);
    };
    CreateUsageLimitCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryCreateUsageLimitCommand(output, context);
    };
    return CreateUsageLimitCommand;
}($Command));
export { CreateUsageLimitCommand };
