// @aws-sdk/client-redshift/dist-es/commands/CreateHsmClientCertificateCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { CreateHsmClientCertificateMessageFilterSensitiveLog, CreateHsmClientCertificateResultFilterSensitiveLog, } from "../models/models_0";
import { deserializeAws_queryCreateHsmClientCertificateCommand, serializeAws_queryCreateHsmClientCertificateCommand, } from "../protocols/Aws_query";
var CreateHsmClientCertificateCommand = (function (_super) {
    __extends(CreateHsmClientCertificateCommand, _super);
    function CreateHsmClientCertificateCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    CreateHsmClientCertificateCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "RedshiftClient";
        var commandName = "CreateHsmClientCertificateCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: CreateHsmClientCertificateMessageFilterSensitiveLog,
            outputFilterSensitiveLog: CreateHsmClientCertificateResultFilterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    CreateHsmClientCertificateCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryCreateHsmClientCertificateCommand(input, context);
    };
    CreateHsmClientCertificateCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryCreateHsmClientCertificateCommand(output, context);
    };
    return CreateHsmClientCertificateCommand;
}($Command));
export { CreateHsmClientCertificateCommand };
