// @aws-sdk/client-redshift/dist-es/commands/DeleteScheduledActionCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DeleteScheduledActionMessageFilterSensitiveLog } from "../models/models_0";
import { deserializeAws_queryDeleteScheduledActionCommand, serializeAws_queryDeleteScheduledActionCommand, } from "../protocols/Aws_query";
var DeleteScheduledActionCommand = (function (_super) {
    __extends(DeleteScheduledActionCommand, _super);
    function DeleteScheduledActionCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DeleteScheduledActionCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "RedshiftClient";
        var commandName = "DeleteScheduledActionCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DeleteScheduledActionMessageFilterSensitiveLog,
            outputFilterSensitiveLog: function (output) { return output; },
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DeleteScheduledActionCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryDeleteScheduledActionCommand(input, context);
    };
    DeleteScheduledActionCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryDeleteScheduledActionCommand(output, context);
    };
    return DeleteScheduledActionCommand;
}($Command));
export { DeleteScheduledActionCommand };
