// @aws-sdk/client-redshift/dist-es/commands/DeleteAuthenticationProfileCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DeleteAuthenticationProfileMessageFilterSensitiveLog, DeleteAuthenticationProfileResultFilterSensitiveLog, } from "../models/models_0";
import { deserializeAws_queryDeleteAuthenticationProfileCommand, serializeAws_queryDeleteAuthenticationProfileCommand, } from "../protocols/Aws_query";
var DeleteAuthenticationProfileCommand = (function (_super) {
    __extends(DeleteAuthenticationProfileCommand, _super);
    function DeleteAuthenticationProfileCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DeleteAuthenticationProfileCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "RedshiftClient";
        var commandName = "DeleteAuthenticationProfileCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DeleteAuthenticationProfileMessageFilterSensitiveLog,
            outputFilterSensitiveLog: DeleteAuthenticationProfileResultFilterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DeleteAuthenticationProfileCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryDeleteAuthenticationProfileCommand(input, context);
    };
    DeleteAuthenticationProfileCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryDeleteAuthenticationProfileCommand(output, context);
    };
    return DeleteAuthenticationProfileCommand;
}($Command));
export { DeleteAuthenticationProfileCommand };
