// @aws-sdk/client-redshift/dist-es/commands/DeleteUsageLimitCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DeleteUsageLimitMessageFilterSensitiveLog } from "../models/models_0";
import { deserializeAws_queryDeleteUsageLimitCommand, serializeAws_queryDeleteUsageLimitCommand, } from "../protocols/Aws_query";
var DeleteUsageLimitCommand = (function (_super) {
    __extends(DeleteUsageLimitCommand, _super);
    function DeleteUsageLimitCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DeleteUsageLimitCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "RedshiftClient";
        var commandName = "DeleteUsageLimitCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DeleteUsageLimitMessageFilterSensitiveLog,
            outputFilterSensitiveLog: function (output) { return output; },
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DeleteUsageLimitCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryDeleteUsageLimitCommand(input, context);
    };
    DeleteUsageLimitCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryDeleteUsageLimitCommand(output, context);
    };
    return DeleteUsageLimitCommand;
}($Command));
export { DeleteUsageLimitCommand };
