// @aws-sdk/client-redshift/dist-es/commands/DeleteHsmConfigurationCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DeleteHsmConfigurationMessageFilterSensitiveLog } from "../models/models_0";
import { deserializeAws_queryDeleteHsmConfigurationCommand, serializeAws_queryDeleteHsmConfigurationCommand, } from "../protocols/Aws_query";
var DeleteHsmConfigurationCommand = (function (_super) {
    __extends(DeleteHsmConfigurationCommand, _super);
    function DeleteHsmConfigurationCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DeleteHsmConfigurationCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "RedshiftClient";
        var commandName = "DeleteHsmConfigurationCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DeleteHsmConfigurationMessageFilterSensitiveLog,
            outputFilterSensitiveLog: function (output) { return output; },
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DeleteHsmConfigurationCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryDeleteHsmConfigurationCommand(input, context);
    };
    DeleteHsmConfigurationCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryDeleteHsmConfigurationCommand(output, context);
    };
    return DeleteHsmConfigurationCommand;
}($Command));
export { DeleteHsmConfigurationCommand };
