// @aws-sdk/client-redshift/dist-es/commands/DeletePartnerCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { PartnerIntegrationInputMessageFilterSensitiveLog, PartnerIntegrationOutputMessageFilterSensitiveLog, } from "../models/models_0";
import { deserializeAws_queryDeletePartnerCommand, serializeAws_queryDeletePartnerCommand, } from "../protocols/Aws_query";
var DeletePartnerCommand = (function (_super) {
    __extends(DeletePartnerCommand, _super);
    function DeletePartnerCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DeletePartnerCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "RedshiftClient";
        var commandName = "DeletePartnerCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: PartnerIntegrationInputMessageFilterSensitiveLog,
            outputFilterSensitiveLog: PartnerIntegrationOutputMessageFilterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DeletePartnerCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryDeletePartnerCommand(input, context);
    };
    DeletePartnerCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryDeletePartnerCommand(output, context);
    };
    return DeletePartnerCommand;
}($Command));
export { DeletePartnerCommand };
