// @aws-sdk/client-redshift/dist-es/commands/DeleteClusterSnapshotCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DeleteClusterSnapshotMessageFilterSensitiveLog, DeleteClusterSnapshotResultFilterSensitiveLog, } from "../models/models_0";
import { deserializeAws_queryDeleteClusterSnapshotCommand, serializeAws_queryDeleteClusterSnapshotCommand, } from "../protocols/Aws_query";
var DeleteClusterSnapshotCommand = (function (_super) {
    __extends(DeleteClusterSnapshotCommand, _super);
    function DeleteClusterSnapshotCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DeleteClusterSnapshotCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "RedshiftClient";
        var commandName = "DeleteClusterSnapshotCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DeleteClusterSnapshotMessageFilterSensitiveLog,
            outputFilterSensitiveLog: DeleteClusterSnapshotResultFilterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DeleteClusterSnapshotCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryDeleteClusterSnapshotCommand(input, context);
    };
    DeleteClusterSnapshotCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryDeleteClusterSnapshotCommand(output, context);
    };
    return DeleteClusterSnapshotCommand;
}($Command));
export { DeleteClusterSnapshotCommand };
