// @aws-sdk/client-redshift/dist-es/commands/DescribeReservedNodesCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DescribeReservedNodesMessageFilterSensitiveLog, ReservedNodesMessageFilterSensitiveLog, } from "../models/models_1";
import { deserializeAws_queryDescribeReservedNodesCommand, serializeAws_queryDescribeReservedNodesCommand, } from "../protocols/Aws_query";
var DescribeReservedNodesCommand = (function (_super) {
    __extends(DescribeReservedNodesCommand, _super);
    function DescribeReservedNodesCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DescribeReservedNodesCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "RedshiftClient";
        var commandName = "DescribeReservedNodesCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DescribeReservedNodesMessageFilterSensitiveLog,
            outputFilterSensitiveLog: ReservedNodesMessageFilterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DescribeReservedNodesCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryDescribeReservedNodesCommand(input, context);
    };
    DescribeReservedNodesCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryDescribeReservedNodesCommand(output, context);
    };
    return DescribeReservedNodesCommand;
}($Command));
export { DescribeReservedNodesCommand };
