// @aws-sdk/client-redshift/dist-es/commands/DescribeSnapshotSchedulesCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DescribeSnapshotSchedulesMessageFilterSensitiveLog, DescribeSnapshotSchedulesOutputMessageFilterSensitiveLog, } from "../models/models_1";
import { deserializeAws_queryDescribeSnapshotSchedulesCommand, serializeAws_queryDescribeSnapshotSchedulesCommand, } from "../protocols/Aws_query";
var DescribeSnapshotSchedulesCommand = (function (_super) {
    __extends(DescribeSnapshotSchedulesCommand, _super);
    function DescribeSnapshotSchedulesCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DescribeSnapshotSchedulesCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "RedshiftClient";
        var commandName = "DescribeSnapshotSchedulesCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DescribeSnapshotSchedulesMessageFilterSensitiveLog,
            outputFilterSensitiveLog: DescribeSnapshotSchedulesOutputMessageFilterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DescribeSnapshotSchedulesCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryDescribeSnapshotSchedulesCommand(input, context);
    };
    DescribeSnapshotSchedulesCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryDescribeSnapshotSchedulesCommand(output, context);
    };
    return DescribeSnapshotSchedulesCommand;
}($Command));
export { DescribeSnapshotSchedulesCommand };
