// @aws-sdk/client-redshift/dist-es/commands/DescribeSnapshotCopyGrantsCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DescribeSnapshotCopyGrantsMessageFilterSensitiveLog, SnapshotCopyGrantMessageFilterSensitiveLog, } from "../models/models_1";
import { deserializeAws_queryDescribeSnapshotCopyGrantsCommand, serializeAws_queryDescribeSnapshotCopyGrantsCommand, } from "../protocols/Aws_query";
var DescribeSnapshotCopyGrantsCommand = (function (_super) {
    __extends(DescribeSnapshotCopyGrantsCommand, _super);
    function DescribeSnapshotCopyGrantsCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DescribeSnapshotCopyGrantsCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "RedshiftClient";
        var commandName = "DescribeSnapshotCopyGrantsCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DescribeSnapshotCopyGrantsMessageFilterSensitiveLog,
            outputFilterSensitiveLog: SnapshotCopyGrantMessageFilterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DescribeSnapshotCopyGrantsCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryDescribeSnapshotCopyGrantsCommand(input, context);
    };
    DescribeSnapshotCopyGrantsCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryDescribeSnapshotCopyGrantsCommand(output, context);
    };
    return DescribeSnapshotCopyGrantsCommand;
}($Command));
export { DescribeSnapshotCopyGrantsCommand };
