// @aws-sdk/client-redshift/dist-es/commands/DescribeHsmConfigurationsCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DescribeHsmConfigurationsMessageFilterSensitiveLog, HsmConfigurationMessageFilterSensitiveLog, } from "../models/models_1";
import { deserializeAws_queryDescribeHsmConfigurationsCommand, serializeAws_queryDescribeHsmConfigurationsCommand, } from "../protocols/Aws_query";
var DescribeHsmConfigurationsCommand = (function (_super) {
    __extends(DescribeHsmConfigurationsCommand, _super);
    function DescribeHsmConfigurationsCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DescribeHsmConfigurationsCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "RedshiftClient";
        var commandName = "DescribeHsmConfigurationsCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DescribeHsmConfigurationsMessageFilterSensitiveLog,
            outputFilterSensitiveLog: HsmConfigurationMessageFilterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DescribeHsmConfigurationsCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryDescribeHsmConfigurationsCommand(input, context);
    };
    DescribeHsmConfigurationsCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryDescribeHsmConfigurationsCommand(output, context);
    };
    return DescribeHsmConfigurationsCommand;
}($Command));
export { DescribeHsmConfigurationsCommand };
