// @aws-sdk/client-redshift/dist-es/commands/DescribeDefaultClusterParametersCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DescribeDefaultClusterParametersMessageFilterSensitiveLog, DescribeDefaultClusterParametersResultFilterSensitiveLog, } from "../models/models_0";
import { deserializeAws_queryDescribeDefaultClusterParametersCommand, serializeAws_queryDescribeDefaultClusterParametersCommand, } from "../protocols/Aws_query";
var DescribeDefaultClusterParametersCommand = (function (_super) {
    __extends(DescribeDefaultClusterParametersCommand, _super);
    function DescribeDefaultClusterParametersCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DescribeDefaultClusterParametersCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "RedshiftClient";
        var commandName = "DescribeDefaultClusterParametersCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DescribeDefaultClusterParametersMessageFilterSensitiveLog,
            outputFilterSensitiveLog: DescribeDefaultClusterParametersResultFilterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DescribeDefaultClusterParametersCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryDescribeDefaultClusterParametersCommand(input, context);
    };
    DescribeDefaultClusterParametersCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryDescribeDefaultClusterParametersCommand(output, context);
    };
    return DescribeDefaultClusterParametersCommand;
}($Command));
export { DescribeDefaultClusterParametersCommand };
