// @aws-sdk/client-redshift/dist-es/commands/DescribeClusterDbRevisionsCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { ClusterDbRevisionsMessageFilterSensitiveLog, DescribeClusterDbRevisionsMessageFilterSensitiveLog, } from "../models/models_0";
import { deserializeAws_queryDescribeClusterDbRevisionsCommand, serializeAws_queryDescribeClusterDbRevisionsCommand, } from "../protocols/Aws_query";
var DescribeClusterDbRevisionsCommand = (function (_super) {
    __extends(DescribeClusterDbRevisionsCommand, _super);
    function DescribeClusterDbRevisionsCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DescribeClusterDbRevisionsCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "RedshiftClient";
        var commandName = "DescribeClusterDbRevisionsCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DescribeClusterDbRevisionsMessageFilterSensitiveLog,
            outputFilterSensitiveLog: ClusterDbRevisionsMessageFilterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DescribeClusterDbRevisionsCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryDescribeClusterDbRevisionsCommand(input, context);
    };
    DescribeClusterDbRevisionsCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryDescribeClusterDbRevisionsCommand(output, context);
    };
    return DescribeClusterDbRevisionsCommand;
}($Command));
export { DescribeClusterDbRevisionsCommand };
