// @aws-sdk/client-redshift/dist-es/commands/DescribeStorageCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { CustomerStorageMessageFilterSensitiveLog } from "../models/models_0";
import { deserializeAws_queryDescribeStorageCommand, serializeAws_queryDescribeStorageCommand, } from "../protocols/Aws_query";
var DescribeStorageCommand = (function (_super) {
    __extends(DescribeStorageCommand, _super);
    function DescribeStorageCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DescribeStorageCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "RedshiftClient";
        var commandName = "DescribeStorageCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: function (input) { return input; },
            outputFilterSensitiveLog: CustomerStorageMessageFilterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DescribeStorageCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryDescribeStorageCommand(input, context);
    };
    DescribeStorageCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryDescribeStorageCommand(output, context);
    };
    return DescribeStorageCommand;
}($Command));
export { DescribeStorageCommand };
