// @aws-sdk/client-redshift/dist-es/commands/DescribeEndpointAccessCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DescribeEndpointAccessMessageFilterSensitiveLog, EndpointAccessListFilterSensitiveLog, } from "../models/models_0";
import { deserializeAws_queryDescribeEndpointAccessCommand, serializeAws_queryDescribeEndpointAccessCommand, } from "../protocols/Aws_query";
var DescribeEndpointAccessCommand = (function (_super) {
    __extends(DescribeEndpointAccessCommand, _super);
    function DescribeEndpointAccessCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DescribeEndpointAccessCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "RedshiftClient";
        var commandName = "DescribeEndpointAccessCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DescribeEndpointAccessMessageFilterSensitiveLog,
            outputFilterSensitiveLog: EndpointAccessListFilterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DescribeEndpointAccessCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryDescribeEndpointAccessCommand(input, context);
    };
    DescribeEndpointAccessCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryDescribeEndpointAccessCommand(output, context);
    };
    return DescribeEndpointAccessCommand;
}($Command));
export { DescribeEndpointAccessCommand };
