// @aws-sdk/client-redshift/dist-es/commands/DescribeClusterSnapshotsCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DescribeClusterSnapshotsMessageFilterSensitiveLog, SnapshotMessageFilterSensitiveLog, } from "../models/models_0";
import { deserializeAws_queryDescribeClusterSnapshotsCommand, serializeAws_queryDescribeClusterSnapshotsCommand, } from "../protocols/Aws_query";
var DescribeClusterSnapshotsCommand = (function (_super) {
    __extends(DescribeClusterSnapshotsCommand, _super);
    function DescribeClusterSnapshotsCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DescribeClusterSnapshotsCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "RedshiftClient";
        var commandName = "DescribeClusterSnapshotsCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DescribeClusterSnapshotsMessageFilterSensitiveLog,
            outputFilterSensitiveLog: SnapshotMessageFilterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DescribeClusterSnapshotsCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryDescribeClusterSnapshotsCommand(input, context);
    };
    DescribeClusterSnapshotsCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryDescribeClusterSnapshotsCommand(output, context);
    };
    return DescribeClusterSnapshotsCommand;
}($Command));
export { DescribeClusterSnapshotsCommand };
