// @aws-sdk/client-redshift/dist-es/commands/DescribeOrderableClusterOptionsCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DescribeOrderableClusterOptionsMessageFilterSensitiveLog, OrderableClusterOptionsMessageFilterSensitiveLog, } from "../models/models_1";
import { deserializeAws_queryDescribeOrderableClusterOptionsCommand, serializeAws_queryDescribeOrderableClusterOptionsCommand, } from "../protocols/Aws_query";
var DescribeOrderableClusterOptionsCommand = (function (_super) {
    __extends(DescribeOrderableClusterOptionsCommand, _super);
    function DescribeOrderableClusterOptionsCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DescribeOrderableClusterOptionsCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "RedshiftClient";
        var commandName = "DescribeOrderableClusterOptionsCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DescribeOrderableClusterOptionsMessageFilterSensitiveLog,
            outputFilterSensitiveLog: OrderableClusterOptionsMessageFilterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DescribeOrderableClusterOptionsCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryDescribeOrderableClusterOptionsCommand(input, context);
    };
    DescribeOrderableClusterOptionsCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryDescribeOrderableClusterOptionsCommand(output, context);
    };
    return DescribeOrderableClusterOptionsCommand;
}($Command));
export { DescribeOrderableClusterOptionsCommand };
