// @aws-sdk/client-redshift/dist-es/commands/DescribeClustersCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { ClustersMessageFilterSensitiveLog, DescribeClustersMessageFilterSensitiveLog, } from "../models/models_0";
import { deserializeAws_queryDescribeClustersCommand, serializeAws_queryDescribeClustersCommand, } from "../protocols/Aws_query";
var DescribeClustersCommand = (function (_super) {
    __extends(DescribeClustersCommand, _super);
    function DescribeClustersCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DescribeClustersCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "RedshiftClient";
        var commandName = "DescribeClustersCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DescribeClustersMessageFilterSensitiveLog,
            outputFilterSensitiveLog: ClustersMessageFilterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DescribeClustersCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryDescribeClustersCommand(input, context);
    };
    DescribeClustersCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryDescribeClustersCommand(output, context);
    };
    return DescribeClustersCommand;
}($Command));
export { DescribeClustersCommand };
