// @aws-sdk/client-redshift/dist-es/commands/DescribeClusterParameterGroupsCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { ClusterParameterGroupsMessageFilterSensitiveLog, DescribeClusterParameterGroupsMessageFilterSensitiveLog, } from "../models/models_0";
import { deserializeAws_queryDescribeClusterParameterGroupsCommand, serializeAws_queryDescribeClusterParameterGroupsCommand, } from "../protocols/Aws_query";
var DescribeClusterParameterGroupsCommand = (function (_super) {
    __extends(DescribeClusterParameterGroupsCommand, _super);
    function DescribeClusterParameterGroupsCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DescribeClusterParameterGroupsCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "RedshiftClient";
        var commandName = "DescribeClusterParameterGroupsCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DescribeClusterParameterGroupsMessageFilterSensitiveLog,
            outputFilterSensitiveLog: ClusterParameterGroupsMessageFilterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DescribeClusterParameterGroupsCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryDescribeClusterParameterGroupsCommand(input, context);
    };
    DescribeClusterParameterGroupsCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryDescribeClusterParameterGroupsCommand(output, context);
    };
    return DescribeClusterParameterGroupsCommand;
}($Command));
export { DescribeClusterParameterGroupsCommand };
