// @aws-sdk/client-redshift/dist-es/commands/DescribeNodeConfigurationOptionsCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DescribeNodeConfigurationOptionsMessageFilterSensitiveLog, NodeConfigurationOptionsMessageFilterSensitiveLog, } from "../models/models_1";
import { deserializeAws_queryDescribeNodeConfigurationOptionsCommand, serializeAws_queryDescribeNodeConfigurationOptionsCommand, } from "../protocols/Aws_query";
var DescribeNodeConfigurationOptionsCommand = (function (_super) {
    __extends(DescribeNodeConfigurationOptionsCommand, _super);
    function DescribeNodeConfigurationOptionsCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DescribeNodeConfigurationOptionsCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "RedshiftClient";
        var commandName = "DescribeNodeConfigurationOptionsCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DescribeNodeConfigurationOptionsMessageFilterSensitiveLog,
            outputFilterSensitiveLog: NodeConfigurationOptionsMessageFilterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DescribeNodeConfigurationOptionsCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryDescribeNodeConfigurationOptionsCommand(input, context);
    };
    DescribeNodeConfigurationOptionsCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryDescribeNodeConfigurationOptionsCommand(output, context);
    };
    return DescribeNodeConfigurationOptionsCommand;
}($Command));
export { DescribeNodeConfigurationOptionsCommand };
