// @aws-sdk/client-redshift/dist-es/commands/DisableLoggingCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DisableLoggingMessageFilterSensitiveLog, LoggingStatusFilterSensitiveLog, } from "../models/models_1";
import { deserializeAws_queryDisableLoggingCommand, serializeAws_queryDisableLoggingCommand, } from "../protocols/Aws_query";
var DisableLoggingCommand = (function (_super) {
    __extends(DisableLoggingCommand, _super);
    function DisableLoggingCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DisableLoggingCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "RedshiftClient";
        var commandName = "DisableLoggingCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DisableLoggingMessageFilterSensitiveLog,
            outputFilterSensitiveLog: LoggingStatusFilterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DisableLoggingCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryDisableLoggingCommand(input, context);
    };
    DisableLoggingCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryDisableLoggingCommand(output, context);
    };
    return DisableLoggingCommand;
}($Command));
export { DisableLoggingCommand };
