// @aws-sdk/client-redshift/dist-es/commands/DisableSnapshotCopyCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DisableSnapshotCopyMessageFilterSensitiveLog, DisableSnapshotCopyResultFilterSensitiveLog, } from "../models/models_1";
import { deserializeAws_queryDisableSnapshotCopyCommand, serializeAws_queryDisableSnapshotCopyCommand, } from "../protocols/Aws_query";
var DisableSnapshotCopyCommand = (function (_super) {
    __extends(DisableSnapshotCopyCommand, _super);
    function DisableSnapshotCopyCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DisableSnapshotCopyCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "RedshiftClient";
        var commandName = "DisableSnapshotCopyCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DisableSnapshotCopyMessageFilterSensitiveLog,
            outputFilterSensitiveLog: DisableSnapshotCopyResultFilterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DisableSnapshotCopyCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryDisableSnapshotCopyCommand(input, context);
    };
    DisableSnapshotCopyCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryDisableSnapshotCopyCommand(output, context);
    };
    return DisableSnapshotCopyCommand;
}($Command));
export { DisableSnapshotCopyCommand };
