// @aws-sdk/client-redshift/dist-es/commands/DisassociateDataShareConsumerCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DataShareFilterSensitiveLog } from "../models/models_0";
import { DisassociateDataShareConsumerMessageFilterSensitiveLog, } from "../models/models_1";
import { deserializeAws_queryDisassociateDataShareConsumerCommand, serializeAws_queryDisassociateDataShareConsumerCommand, } from "../protocols/Aws_query";
var DisassociateDataShareConsumerCommand = (function (_super) {
    __extends(DisassociateDataShareConsumerCommand, _super);
    function DisassociateDataShareConsumerCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DisassociateDataShareConsumerCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "RedshiftClient";
        var commandName = "DisassociateDataShareConsumerCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DisassociateDataShareConsumerMessageFilterSensitiveLog,
            outputFilterSensitiveLog: DataShareFilterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DisassociateDataShareConsumerCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryDisassociateDataShareConsumerCommand(input, context);
    };
    DisassociateDataShareConsumerCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryDisassociateDataShareConsumerCommand(output, context);
    };
    return DisassociateDataShareConsumerCommand;
}($Command));
export { DisassociateDataShareConsumerCommand };
