// @aws-sdk/client-redshift/dist-es/commands/EnableSnapshotCopyCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { EnableSnapshotCopyMessageFilterSensitiveLog, EnableSnapshotCopyResultFilterSensitiveLog, } from "../models/models_1";
import { deserializeAws_queryEnableSnapshotCopyCommand, serializeAws_queryEnableSnapshotCopyCommand, } from "../protocols/Aws_query";
var EnableSnapshotCopyCommand = (function (_super) {
    __extends(EnableSnapshotCopyCommand, _super);
    function EnableSnapshotCopyCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    EnableSnapshotCopyCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "RedshiftClient";
        var commandName = "EnableSnapshotCopyCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: EnableSnapshotCopyMessageFilterSensitiveLog,
            outputFilterSensitiveLog: EnableSnapshotCopyResultFilterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    EnableSnapshotCopyCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryEnableSnapshotCopyCommand(input, context);
    };
    EnableSnapshotCopyCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryEnableSnapshotCopyCommand(output, context);
    };
    return EnableSnapshotCopyCommand;
}($Command));
export { EnableSnapshotCopyCommand };
