// @aws-sdk/client-redshift/dist-es/commands/EnableLoggingCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { EnableLoggingMessageFilterSensitiveLog, LoggingStatusFilterSensitiveLog, } from "../models/models_1";
import { deserializeAws_queryEnableLoggingCommand, serializeAws_queryEnableLoggingCommand, } from "../protocols/Aws_query";
var EnableLoggingCommand = (function (_super) {
    __extends(EnableLoggingCommand, _super);
    function EnableLoggingCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    EnableLoggingCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "RedshiftClient";
        var commandName = "EnableLoggingCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: EnableLoggingMessageFilterSensitiveLog,
            outputFilterSensitiveLog: LoggingStatusFilterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    EnableLoggingCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryEnableLoggingCommand(input, context);
    };
    EnableLoggingCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryEnableLoggingCommand(output, context);
    };
    return EnableLoggingCommand;
}($Command));
export { EnableLoggingCommand };
