// @aws-sdk/client-redshift/dist-es/commands/GetClusterCredentialsWithIAMCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { ClusterExtendedCredentialsFilterSensitiveLog } from "../models/models_0";
import { GetClusterCredentialsWithIAMMessageFilterSensitiveLog, } from "../models/models_1";
import { deserializeAws_queryGetClusterCredentialsWithIAMCommand, serializeAws_queryGetClusterCredentialsWithIAMCommand, } from "../protocols/Aws_query";
var GetClusterCredentialsWithIAMCommand = (function (_super) {
    __extends(GetClusterCredentialsWithIAMCommand, _super);
    function GetClusterCredentialsWithIAMCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    GetClusterCredentialsWithIAMCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "RedshiftClient";
        var commandName = "GetClusterCredentialsWithIAMCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: GetClusterCredentialsWithIAMMessageFilterSensitiveLog,
            outputFilterSensitiveLog: ClusterExtendedCredentialsFilterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    GetClusterCredentialsWithIAMCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryGetClusterCredentialsWithIAMCommand(input, context);
    };
    GetClusterCredentialsWithIAMCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryGetClusterCredentialsWithIAMCommand(output, context);
    };
    return GetClusterCredentialsWithIAMCommand;
}($Command));
export { GetClusterCredentialsWithIAMCommand };
