// @aws-sdk/client-redshift/dist-es/commands/GetReservedNodeExchangeOfferingsCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { GetReservedNodeExchangeOfferingsInputMessageFilterSensitiveLog, GetReservedNodeExchangeOfferingsOutputMessageFilterSensitiveLog, } from "../models/models_1";
import { deserializeAws_queryGetReservedNodeExchangeOfferingsCommand, serializeAws_queryGetReservedNodeExchangeOfferingsCommand, } from "../protocols/Aws_query";
var GetReservedNodeExchangeOfferingsCommand = (function (_super) {
    __extends(GetReservedNodeExchangeOfferingsCommand, _super);
    function GetReservedNodeExchangeOfferingsCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    GetReservedNodeExchangeOfferingsCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "RedshiftClient";
        var commandName = "GetReservedNodeExchangeOfferingsCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: GetReservedNodeExchangeOfferingsInputMessageFilterSensitiveLog,
            outputFilterSensitiveLog: GetReservedNodeExchangeOfferingsOutputMessageFilterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    GetReservedNodeExchangeOfferingsCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryGetReservedNodeExchangeOfferingsCommand(input, context);
    };
    GetReservedNodeExchangeOfferingsCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryGetReservedNodeExchangeOfferingsCommand(output, context);
    };
    return GetReservedNodeExchangeOfferingsCommand;
}($Command));
export { GetReservedNodeExchangeOfferingsCommand };
