// @aws-sdk/client-redshift/dist-es/commands/GetReservedNodeExchangeConfigurationOptionsCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { GetReservedNodeExchangeConfigurationOptionsInputMessageFilterSensitiveLog, GetReservedNodeExchangeConfigurationOptionsOutputMessageFilterSensitiveLog, } from "../models/models_1";
import { deserializeAws_queryGetReservedNodeExchangeConfigurationOptionsCommand, serializeAws_queryGetReservedNodeExchangeConfigurationOptionsCommand, } from "../protocols/Aws_query";
var GetReservedNodeExchangeConfigurationOptionsCommand = (function (_super) {
    __extends(GetReservedNodeExchangeConfigurationOptionsCommand, _super);
    function GetReservedNodeExchangeConfigurationOptionsCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    GetReservedNodeExchangeConfigurationOptionsCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "RedshiftClient";
        var commandName = "GetReservedNodeExchangeConfigurationOptionsCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: GetReservedNodeExchangeConfigurationOptionsInputMessageFilterSensitiveLog,
            outputFilterSensitiveLog: GetReservedNodeExchangeConfigurationOptionsOutputMessageFilterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    GetReservedNodeExchangeConfigurationOptionsCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryGetReservedNodeExchangeConfigurationOptionsCommand(input, context);
    };
    GetReservedNodeExchangeConfigurationOptionsCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryGetReservedNodeExchangeConfigurationOptionsCommand(output, context);
    };
    return GetReservedNodeExchangeConfigurationOptionsCommand;
}($Command));
export { GetReservedNodeExchangeConfigurationOptionsCommand };
