// @aws-sdk/client-redshift/dist-es/commands/ModifySnapshotCopyRetentionPeriodCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { ModifySnapshotCopyRetentionPeriodMessageFilterSensitiveLog, ModifySnapshotCopyRetentionPeriodResultFilterSensitiveLog, } from "../models/models_1";
import { deserializeAws_queryModifySnapshotCopyRetentionPeriodCommand, serializeAws_queryModifySnapshotCopyRetentionPeriodCommand, } from "../protocols/Aws_query";
var ModifySnapshotCopyRetentionPeriodCommand = (function (_super) {
    __extends(ModifySnapshotCopyRetentionPeriodCommand, _super);
    function ModifySnapshotCopyRetentionPeriodCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    ModifySnapshotCopyRetentionPeriodCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "RedshiftClient";
        var commandName = "ModifySnapshotCopyRetentionPeriodCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: ModifySnapshotCopyRetentionPeriodMessageFilterSensitiveLog,
            outputFilterSensitiveLog: ModifySnapshotCopyRetentionPeriodResultFilterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    ModifySnapshotCopyRetentionPeriodCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryModifySnapshotCopyRetentionPeriodCommand(input, context);
    };
    ModifySnapshotCopyRetentionPeriodCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryModifySnapshotCopyRetentionPeriodCommand(output, context);
    };
    return ModifySnapshotCopyRetentionPeriodCommand;
}($Command));
export { ModifySnapshotCopyRetentionPeriodCommand };
