// @aws-sdk/client-redshift/dist-es/commands/ModifyClusterCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { ModifyClusterMessageFilterSensitiveLog, ModifyClusterResultFilterSensitiveLog, } from "../models/models_1";
import { deserializeAws_queryModifyClusterCommand, serializeAws_queryModifyClusterCommand, } from "../protocols/Aws_query";
var ModifyClusterCommand = (function (_super) {
    __extends(ModifyClusterCommand, _super);
    function ModifyClusterCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    ModifyClusterCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "RedshiftClient";
        var commandName = "ModifyClusterCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: ModifyClusterMessageFilterSensitiveLog,
            outputFilterSensitiveLog: ModifyClusterResultFilterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    ModifyClusterCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryModifyClusterCommand(input, context);
    };
    ModifyClusterCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryModifyClusterCommand(output, context);
    };
    return ModifyClusterCommand;
}($Command));
export { ModifyClusterCommand };
