// @aws-sdk/client-redshift/dist-es/commands/ModifySnapshotScheduleCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { SnapshotScheduleFilterSensitiveLog } from "../models/models_0";
import { ModifySnapshotScheduleMessageFilterSensitiveLog } from "../models/models_1";
import { deserializeAws_queryModifySnapshotScheduleCommand, serializeAws_queryModifySnapshotScheduleCommand, } from "../protocols/Aws_query";
var ModifySnapshotScheduleCommand = (function (_super) {
    __extends(ModifySnapshotScheduleCommand, _super);
    function ModifySnapshotScheduleCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    ModifySnapshotScheduleCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "RedshiftClient";
        var commandName = "ModifySnapshotScheduleCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: ModifySnapshotScheduleMessageFilterSensitiveLog,
            outputFilterSensitiveLog: SnapshotScheduleFilterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    ModifySnapshotScheduleCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryModifySnapshotScheduleCommand(input, context);
    };
    ModifySnapshotScheduleCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryModifySnapshotScheduleCommand(output, context);
    };
    return ModifySnapshotScheduleCommand;
}($Command));
export { ModifySnapshotScheduleCommand };
