// @aws-sdk/client-redshift/dist-es/commands/ModifyScheduledActionCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { ScheduledActionFilterSensitiveLog } from "../models/models_0";
import { ModifyScheduledActionMessageFilterSensitiveLog } from "../models/models_1";
import { deserializeAws_queryModifyScheduledActionCommand, serializeAws_queryModifyScheduledActionCommand, } from "../protocols/Aws_query";
var ModifyScheduledActionCommand = (function (_super) {
    __extends(ModifyScheduledActionCommand, _super);
    function ModifyScheduledActionCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    ModifyScheduledActionCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "RedshiftClient";
        var commandName = "ModifyScheduledActionCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: ModifyScheduledActionMessageFilterSensitiveLog,
            outputFilterSensitiveLog: ScheduledActionFilterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    ModifyScheduledActionCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryModifyScheduledActionCommand(input, context);
    };
    ModifyScheduledActionCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryModifyScheduledActionCommand(output, context);
    };
    return ModifyScheduledActionCommand;
}($Command));
export { ModifyScheduledActionCommand };
