// @aws-sdk/client-redshift/dist-es/commands/ModifyAuthenticationProfileCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { ModifyAuthenticationProfileMessageFilterSensitiveLog, ModifyAuthenticationProfileResultFilterSensitiveLog, } from "../models/models_1";
import { deserializeAws_queryModifyAuthenticationProfileCommand, serializeAws_queryModifyAuthenticationProfileCommand, } from "../protocols/Aws_query";
var ModifyAuthenticationProfileCommand = (function (_super) {
    __extends(ModifyAuthenticationProfileCommand, _super);
    function ModifyAuthenticationProfileCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    ModifyAuthenticationProfileCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "RedshiftClient";
        var commandName = "ModifyAuthenticationProfileCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: ModifyAuthenticationProfileMessageFilterSensitiveLog,
            outputFilterSensitiveLog: ModifyAuthenticationProfileResultFilterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    ModifyAuthenticationProfileCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryModifyAuthenticationProfileCommand(input, context);
    };
    ModifyAuthenticationProfileCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryModifyAuthenticationProfileCommand(output, context);
    };
    return ModifyAuthenticationProfileCommand;
}($Command));
export { ModifyAuthenticationProfileCommand };
