// @aws-sdk/client-redshift/dist-es/commands/PurchaseReservedNodeOfferingCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { PurchaseReservedNodeOfferingMessageFilterSensitiveLog, PurchaseReservedNodeOfferingResultFilterSensitiveLog, } from "../models/models_1";
import { deserializeAws_queryPurchaseReservedNodeOfferingCommand, serializeAws_queryPurchaseReservedNodeOfferingCommand, } from "../protocols/Aws_query";
var PurchaseReservedNodeOfferingCommand = (function (_super) {
    __extends(PurchaseReservedNodeOfferingCommand, _super);
    function PurchaseReservedNodeOfferingCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    PurchaseReservedNodeOfferingCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "RedshiftClient";
        var commandName = "PurchaseReservedNodeOfferingCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: PurchaseReservedNodeOfferingMessageFilterSensitiveLog,
            outputFilterSensitiveLog: PurchaseReservedNodeOfferingResultFilterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    PurchaseReservedNodeOfferingCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryPurchaseReservedNodeOfferingCommand(input, context);
    };
    PurchaseReservedNodeOfferingCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryPurchaseReservedNodeOfferingCommand(output, context);
    };
    return PurchaseReservedNodeOfferingCommand;
}($Command));
export { PurchaseReservedNodeOfferingCommand };
