// @aws-sdk/client-redshift/dist-es/commands/RebootClusterCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { RebootClusterMessageFilterSensitiveLog, RebootClusterResultFilterSensitiveLog, } from "../models/models_1";
import { deserializeAws_queryRebootClusterCommand, serializeAws_queryRebootClusterCommand, } from "../protocols/Aws_query";
var RebootClusterCommand = (function (_super) {
    __extends(RebootClusterCommand, _super);
    function RebootClusterCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    RebootClusterCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "RedshiftClient";
        var commandName = "RebootClusterCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: RebootClusterMessageFilterSensitiveLog,
            outputFilterSensitiveLog: RebootClusterResultFilterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    RebootClusterCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryRebootClusterCommand(input, context);
    };
    RebootClusterCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryRebootClusterCommand(output, context);
    };
    return RebootClusterCommand;
}($Command));
export { RebootClusterCommand };
