// @aws-sdk/client-redshift/dist-es/commands/ResizeClusterCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { ResizeClusterMessageFilterSensitiveLog } from "../models/models_0";
import { ResizeClusterResultFilterSensitiveLog } from "../models/models_1";
import { deserializeAws_queryResizeClusterCommand, serializeAws_queryResizeClusterCommand, } from "../protocols/Aws_query";
var ResizeClusterCommand = (function (_super) {
    __extends(ResizeClusterCommand, _super);
    function ResizeClusterCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    ResizeClusterCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "RedshiftClient";
        var commandName = "ResizeClusterCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: ResizeClusterMessageFilterSensitiveLog,
            outputFilterSensitiveLog: ResizeClusterResultFilterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    ResizeClusterCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryResizeClusterCommand(input, context);
    };
    ResizeClusterCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryResizeClusterCommand(output, context);
    };
    return ResizeClusterCommand;
}($Command));
export { ResizeClusterCommand };
