// @aws-sdk/client-redshift/dist-es/commands/RestoreFromClusterSnapshotCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { RestoreFromClusterSnapshotMessageFilterSensitiveLog, RestoreFromClusterSnapshotResultFilterSensitiveLog, } from "../models/models_1";
import { deserializeAws_queryRestoreFromClusterSnapshotCommand, serializeAws_queryRestoreFromClusterSnapshotCommand, } from "../protocols/Aws_query";
var RestoreFromClusterSnapshotCommand = (function (_super) {
    __extends(RestoreFromClusterSnapshotCommand, _super);
    function RestoreFromClusterSnapshotCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    RestoreFromClusterSnapshotCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "RedshiftClient";
        var commandName = "RestoreFromClusterSnapshotCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: RestoreFromClusterSnapshotMessageFilterSensitiveLog,
            outputFilterSensitiveLog: RestoreFromClusterSnapshotResultFilterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    RestoreFromClusterSnapshotCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryRestoreFromClusterSnapshotCommand(input, context);
    };
    RestoreFromClusterSnapshotCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryRestoreFromClusterSnapshotCommand(output, context);
    };
    return RestoreFromClusterSnapshotCommand;
}($Command));
export { RestoreFromClusterSnapshotCommand };
