// @aws-sdk/client-redshift/dist-es/commands/ResumeClusterCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { ResumeClusterMessageFilterSensitiveLog } from "../models/models_0";
import { ResumeClusterResultFilterSensitiveLog } from "../models/models_1";
import { deserializeAws_queryResumeClusterCommand, serializeAws_queryResumeClusterCommand, } from "../protocols/Aws_query";
var ResumeClusterCommand = (function (_super) {
    __extends(ResumeClusterCommand, _super);
    function ResumeClusterCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    ResumeClusterCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "RedshiftClient";
        var commandName = "ResumeClusterCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: ResumeClusterMessageFilterSensitiveLog,
            outputFilterSensitiveLog: ResumeClusterResultFilterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    ResumeClusterCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryResumeClusterCommand(input, context);
    };
    ResumeClusterCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryResumeClusterCommand(output, context);
    };
    return ResumeClusterCommand;
}($Command));
export { ResumeClusterCommand };
