// @aws-sdk/client-redshift/dist-es/commands/RevokeSnapshotAccessCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { RevokeSnapshotAccessMessageFilterSensitiveLog, RevokeSnapshotAccessResultFilterSensitiveLog, } from "../models/models_1";
import { deserializeAws_queryRevokeSnapshotAccessCommand, serializeAws_queryRevokeSnapshotAccessCommand, } from "../protocols/Aws_query";
var RevokeSnapshotAccessCommand = (function (_super) {
    __extends(RevokeSnapshotAccessCommand, _super);
    function RevokeSnapshotAccessCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    RevokeSnapshotAccessCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "RedshiftClient";
        var commandName = "RevokeSnapshotAccessCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: RevokeSnapshotAccessMessageFilterSensitiveLog,
            outputFilterSensitiveLog: RevokeSnapshotAccessResultFilterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    RevokeSnapshotAccessCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryRevokeSnapshotAccessCommand(input, context);
    };
    RevokeSnapshotAccessCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryRevokeSnapshotAccessCommand(output, context);
    };
    return RevokeSnapshotAccessCommand;
}($Command));
export { RevokeSnapshotAccessCommand };
