// @aws-sdk/client-redshift/dist-es/commands/UpdatePartnerStatusCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { PartnerIntegrationOutputMessageFilterSensitiveLog } from "../models/models_0";
import { UpdatePartnerStatusInputMessageFilterSensitiveLog } from "../models/models_1";
import { deserializeAws_queryUpdatePartnerStatusCommand, serializeAws_queryUpdatePartnerStatusCommand, } from "../protocols/Aws_query";
var UpdatePartnerStatusCommand = (function (_super) {
    __extends(UpdatePartnerStatusCommand, _super);
    function UpdatePartnerStatusCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    UpdatePartnerStatusCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "RedshiftClient";
        var commandName = "UpdatePartnerStatusCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: UpdatePartnerStatusInputMessageFilterSensitiveLog,
            outputFilterSensitiveLog: PartnerIntegrationOutputMessageFilterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    UpdatePartnerStatusCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryUpdatePartnerStatusCommand(input, context);
    };
    UpdatePartnerStatusCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryUpdatePartnerStatusCommand(output, context);
    };
    return UpdatePartnerStatusCommand;
}($Command));
export { UpdatePartnerStatusCommand };
