// @aws-sdk/client-redshift/dist-cjs/commands/DescribeClusterDbRevisionsCommand.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.DescribeClusterDbRevisionsCommand = void 0;
const middleware_serde_1 = require("@aws-sdk/middleware-serde");
const smithy_client_1 = require("@aws-sdk/smithy-client");
const models_0_1 = require("../models/models_0");
const Aws_query_1 = require("../protocols/Aws_query");
class DescribeClusterDbRevisionsCommand extends smithy_client_1.Command {
    constructor(input) {
        super();
        this.input = input;
    }
    resolveMiddleware(clientStack, configuration, options) {
        this.middlewareStack.use((0, middleware_serde_1.getSerdePlugin)(configuration, this.serialize, this.deserialize));
        const stack = clientStack.concat(this.middlewareStack);
        const { logger } = configuration;
        const clientName = "RedshiftClient";
        const commandName = "DescribeClusterDbRevisionsCommand";
        const handlerExecutionContext = {
            logger,
            clientName,
            commandName,
            inputFilterSensitiveLog: models_0_1.DescribeClusterDbRevisionsMessageFilterSensitiveLog,
            outputFilterSensitiveLog: models_0_1.ClusterDbRevisionsMessageFilterSensitiveLog,
        };
        const { requestHandler } = configuration;
        return stack.resolve((request) => requestHandler.handle(request.request, options || {}), handlerExecutionContext);
    }
    serialize(input, context) {
        return (0, Aws_query_1.serializeAws_queryDescribeClusterDbRevisionsCommand)(input, context);
    }
    deserialize(output, context) {
        return (0, Aws_query_1.deserializeAws_queryDescribeClusterDbRevisionsCommand)(output, context);
    }
}
exports.DescribeClusterDbRevisionsCommand = DescribeClusterDbRevisionsCommand;
