// @aws-sdk/client-redshift/dist-cjs/commands/DisableSnapshotCopyCommand.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.DisableSnapshotCopyCommand = void 0;
const middleware_serde_1 = require("@aws-sdk/middleware-serde");
const smithy_client_1 = require("@aws-sdk/smithy-client");
const models_1_1 = require("../models/models_1");
const Aws_query_1 = require("../protocols/Aws_query");
class DisableSnapshotCopyCommand extends smithy_client_1.Command {
    constructor(input) {
        super();
        this.input = input;
    }
    resolveMiddleware(clientStack, configuration, options) {
        this.middlewareStack.use((0, middleware_serde_1.getSerdePlugin)(configuration, this.serialize, this.deserialize));
        const stack = clientStack.concat(this.middlewareStack);
        const { logger } = configuration;
        const clientName = "RedshiftClient";
        const commandName = "DisableSnapshotCopyCommand";
        const handlerExecutionContext = {
            logger,
            clientName,
            commandName,
            inputFilterSensitiveLog: models_1_1.DisableSnapshotCopyMessageFilterSensitiveLog,
            outputFilterSensitiveLog: models_1_1.DisableSnapshotCopyResultFilterSensitiveLog,
        };
        const { requestHandler } = configuration;
        return stack.resolve((request) => requestHandler.handle(request.request, options || {}), handlerExecutionContext);
    }
    serialize(input, context) {
        return (0, Aws_query_1.serializeAws_queryDisableSnapshotCopyCommand)(input, context);
    }
    deserialize(output, context) {
        return (0, Aws_query_1.deserializeAws_queryDisableSnapshotCopyCommand)(output, context);
    }
}
exports.DisableSnapshotCopyCommand = DisableSnapshotCopyCommand;
