// @aws-sdk/client-redshift/dist-cjs/commands/ModifyAquaConfigurationCommand.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ModifyAquaConfigurationCommand = void 0;
const middleware_serde_1 = require("@aws-sdk/middleware-serde");
const smithy_client_1 = require("@aws-sdk/smithy-client");
const models_1_1 = require("../models/models_1");
const Aws_query_1 = require("../protocols/Aws_query");
class ModifyAquaConfigurationCommand extends smithy_client_1.Command {
    constructor(input) {
        super();
        this.input = input;
    }
    resolveMiddleware(clientStack, configuration, options) {
        this.middlewareStack.use((0, middleware_serde_1.getSerdePlugin)(configuration, this.serialize, this.deserialize));
        const stack = clientStack.concat(this.middlewareStack);
        const { logger } = configuration;
        const clientName = "RedshiftClient";
        const commandName = "ModifyAquaConfigurationCommand";
        const handlerExecutionContext = {
            logger,
            clientName,
            commandName,
            inputFilterSensitiveLog: models_1_1.ModifyAquaInputMessageFilterSensitiveLog,
            outputFilterSensitiveLog: models_1_1.ModifyAquaOutputMessageFilterSensitiveLog,
        };
        const { requestHandler } = configuration;
        return stack.resolve((request) => requestHandler.handle(request.request, options || {}), handlerExecutionContext);
    }
    serialize(input, context) {
        return (0, Aws_query_1.serializeAws_queryModifyAquaConfigurationCommand)(input, context);
    }
    deserialize(output, context) {
        return (0, Aws_query_1.deserializeAws_queryModifyAquaConfigurationCommand)(output, context);
    }
}
exports.ModifyAquaConfigurationCommand = ModifyAquaConfigurationCommand;
